### DISCUSSION TITLE
How to add a wind turbine with VSG into IEEE-39 system

### DISCUSSION THREAD

**Discussion by Albizzian** [CURENT/andes#512] [Q&A] [answered]
I want to add a wind turbine with VSG into IEEE-39 system, and I found a module named REGCV1 with VSG, but I can’t add it into the power system (displaying some errors, maybe lacking some required modules). I also checked the example system named ieee14_wt3 in ANDES, but its renewable module is REGCA1 without VSG. I would like to ask how to add a renewable module with VSG to provide virtual inertia.

**Comment by jinningwang** [comment]
> displaying some errors, maybe lacking some required modules

Ensure the input file has desired model input.

> I would like to ask how to add a renewable module with VSG to provide virtual inertia.

This might require model development efforts. There is a similar discussion, https://github.com/CURENT/andes/discussions/502

**Reply by Albizzian** [reply]
Thanks for your reply! I add two REGCV1 models into ieee39 system, but I came into this error. Could you tell me what's the problem and how should I adjust the paraments? And here is the original document.
<img width="555" alt="Snipaste_2024-03-09_11-22-09" src="https://github.com/CURENT/andes/assets/78994912/a85b5aa5-34fd-44ba-81d4-4762f52e8c57">
[ieee39_coi_REGCV1.xlsx](https://github.com/CURENT/andes/files/14544760/ieee39_coi_REGCV1.xlsx)

**Reply by jinningwang** [reply, marked as answer]
The error info said TDS initialization failed, due to mismatch on `a` and `v`. It is most likely caused by power balance.

Check the documentation for how to replace a static generator with a dynamic generator: https://docs.andes.app/en/latest/groupdoc/StaticGen.html#staticgen

**Reply by Albizzian** [reply]
Jinning, Thank you very much!  I solved the problem successfully.  The key is to ensure that gammap and gammaq, respectively, of all dynamic power sources sum up to 1.0.

**Reply by ramirotorresy19-cmd** [reply]
Hello Jinning,

I am currently facing a similar issue when integrating a grid-forming inverter (GFM) into the IEEE-39 system. Unfortunately, the documentation you previously shared is no longer available (https://docs.andes.app/en/latest/groupdoc/StaticGen.html#staticgen).

Would you happen to know if there is an updated or alternative documentation that covers the same topic? Any guidance or references you could point me to would be greatly appreciated.

Thank you in advance for your help.

**Reply by jinningwang** [reply]
> Hello Jinning,
> 
> I am currently facing a similar issue when integrating a grid-forming inverter (GFM) into the IEEE-39 system. Unfortunately, the documentation you previously shared is no longer available (https://docs.andes.app/en/latest/groupdoc/StaticGen.html#staticgen).
> 
> Would you happen to know if there is an updated or alternative documentation that covers the same topic? Any guidance or references you could point me to would be greatly appreciated.
> 
> Thank you in advance for your help.

The doc is reorganized in 2.0:
https://docs.andes.app/en/latest/reference/models/StaticGen.html#staticgen

For the model you want, check the existing model library. If there is no model ready to use, then model development efforts are still needed: combine VSG and Wind Turbines.

**Reply by ramirotorresy19-cmd** [reply]
Thank you so much for your help! I really appreciate it. We’ve been trying to resolve this issue for about a month now, so I hope that file will point us in the right direction.

Thanks again, Jinning!

**Reply by cuihantao** [reply]
Did you manage to add the model? You might start with a basic ieee 39 bus system and then replace one synchronous (and its controller) with one converter at a time. After each replacement, run power flow, initialize dynamics, and run small-signal stability to ensure the parameters are correct.
